File: 1884/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(twoEggDrop(100))
}

func twoEggDrop(n int) int {
	f := make([]int, n+1)
	for i := range f {
		f[i] = math.MaxInt32 / 2
	}
	f[0] = 0
	for i := 1; i <= n; i++ {
		for k := 1; k <= i; k++{
			f[i] = min(f[i], max(k-1, f[i-k]) + 1)
		}
	}
	return f[n]
}

File: 1884/main.rs
struct Solution {}

impl Solution {
    pub fn two_egg_drop(n: i32) -> i32 {
        let mut f = vec![i32::MAX/2; (n+1) as usize];
        f[0] = 0;
        for i in 1..=n {
            for k in 1..=i {
                f[i as usize] = std::cmp::min(f[i as usize], std::cmp::max(f[(i-k)as usize], k as i32-1)+1);
            }
        }
        f[n as usize]
    }
}

fn main() {
    println!("{}", Solution::two_egg_drop(100));
}